src/OmnichannelChatSDK.ts

```diff
@@ -2012,7 +1984,27 @@ class OmnichannelChatSDK {
     private async getChatConfig(optionalParams: GetLiveChatConfigOptionalParams = {}): Promise<ChatConfig> {
         const { sendCacheHeaders } = optionalParams;
         const bypassCache = sendCacheHeaders === true;
-        const liveChatConfig = await this.OCClient.getChatConfig(this.requestId, bypassCache);
+
+        let shouldUseFallbackOrgUrl = false;
+        let liveChatConfig;
+
+        try {
+            liveChatConfig = await this.OCClient.getChatConfig(this.requestId, bypassCache);
+        } catch (error) {
+            // Fallback on orgUrl which got converted to Core Services orgUrl
+            if (isCoreServicesOrgUrlDNSError(error, this.coreServicesOrgUrl, this.dynamicsLocationCode)) { // eslint-disable-line @typescript-eslint/no-explicit-any
+                shouldUseFallbackOrgUrl = true
+            } else {
+                throw error // Bubble up error by default to throw ChatConfigRetrievalFailure
+            }
+        }
+
+        if (shouldUseFallbackOrgUrl) {
+            this.omnichannelConfig.orgUrl = this.unqServicesOrgUrl as string;
+            this.OCClient = await OCSDKProvider.getSDK(this.omnichannelConfig as IOmnichannelConfiguration, ocSDKConfiguration as ISDKConfiguration, this.ocSdkLogger as OCSDKLogger);
+            liveChatConfig = await this.OCClient.getChatConfig(this.requestId, bypassCache); // Bubble up error by default to throw ChatConfigRetrievalFailure
```

Comment:
add this to an external function as error handling and call it inside the catch block instead of just setting a flag,  its a biit confussing ,

also , personally I rather use shouldUseFallbackOrgUrl === true